R/busy-indicators.R

```diff
@@ -69,6 +68,16 @@ useBusyIndicators <- function(..., spinners = TRUE, pulse = TRUE) {
 #' this function in the app's UI.
 #'
 #' @param ... Currently ignored (for future expansion).
+#' @param spinner_type The type of spinner to use for the busy indicator. This
+#'   can be any of the spinner types from the `shiny` package.
+#'
+#'   Available spinner types include:
+#'   "`r paste0(.spinner_types, collapse = '", "')`".
+#'
+#'    Alternatively, you can provide a CSS URL to a custom spinner, e.g.
+#'   `../spinners/my-spinning-loader.svg` (use [shiny::addResourcePath()] to
+#'   ensure the `spinner` directory is accessible). Custom URLs must start with
+#'   `.`, `/` or `http`.
```

Comment:
This part is a little tricky and intended for advanced users. There's an example in the demo app in the OP.

It's another "which path, where?" problem in that this URL is relative to hosted location of `busy-indicators.css` because that's where the CSS rules that use the variable are located. 
              
That means the correct URL can depend on how the app is deployed, e.g. in Shiny for R `../custom.svg` would find `www/custom.svg` because that file ends up hosted at `shiny-busy-indicators-1.8.1.9001/busy-indicators.css`. In Shiny for Python, however, dependencies are typically in `lib/{dep_name}/...` so the correct URL is likely `../../custom.svg` (plus adding `www/` to static resources).

On the other hand, you can't use absolute paths without also considering where the app is deployed. e.g. locally `/custom.svg` probably works but it might not work on shinyapps.io if the app is deploy to `user.shinyapps.io/app_name/`.

I wasn't sure how to cram this into a simple sentence.

Reply:
Solution: base64 encode local file paths, don't deal with relative paths online.